README.md

```diff
@@ -1,53 +1,112 @@
 [![node](https://img.shields.io/badge/node-16-brightgreen)](https://nodejs.org/download/release/latest-fermium/)
 
-## Quick setup
+Drive server wip is the new API to Drive based in NestJS and following Clean Arquitecture and DDD(Domain Driven Design).
+## Quick Setup
+
+* Yarn
 
   ```npm i -g yarn```
 
-# Install
+## Table of Contents
+
+- [Install](#how-to-install)
+- [Start the app](#start-app)
+  - [Start with docker](#running-in-docker)
+- [Testing](#testing)
+  - [Unit Testing](#unit-testing)
+  - [End to End Testing](#end-to-end-testing)
+- [Guideline Nest.js](#guideline-nest.js)
+  - [Modules](#modules)
+  - [Externals](#externals)
+- [Swagger](#api-documentation-with-swagger)
+
+## How to Install
 
 - Create a `.npmrc` file from the `.npmrc.template` example provided in the repo. 
 - Replace `TOKEN` with your own [Github Personal Access Token](https://docs.github.com/en/github/authenticating-to-github/keeping-your-account-and-data-secure/creating-a-personal-access-token) with `read:packages` permission **ONLY**
 - Use `yarn` to install project dependencies.
 
-#### Database setup (MariaDB)
+## Start app
 
+Run `yarn start` to start server in production mode.
 
+Run `yarn start:dev` to start with nodemon and development environment.
+### Running in docker:
 
-#### Start app
+With docker-compose:
+```bash
+docker-compose up
+```
 
-Run `yarn start` to start server in production mode.
+## Testing
+### Unit testing
+
+You can run unit tests with:
 
-Run `yarn run dev` to start with nodemon and development environment.
+```bash
+yarn run test
+```
 
-# Running:
+### End to End Testing
 
-With docker-compose:
+Running e2e test requires creating a database first (check .env.test file), and have mariadb started.
 
-`docker-compose up`
+You can either run :
 
-Checkout localhost:3000/api to see every endpoint of the app along documentation
+```bash
+yarn run test:e2e
+```
 
-Locally:
 
-Make sure to have mariadb running in port 6603 and run: `yarn start:dev`
+## Guideline Nest.js
+This project is based in <a href="http://nestjs.com/" target="blank">NestJS</a> and implement DDD (Domain Driven Design).
+Important in DDD is separate this layers:
+- Use cases
+- Persistence
+- Domain
+- Controllers
 
-# Unit testing
+So, following best practices in NestJS we have 2 primary folders:
+- Modules
+- Externals
 
-You can run unit tests with:
+### Modules
+In this folder we have all "use cases" or "context" and following nest.js modules structure, we have 3 concepts of files:
+- <strong>Module:</strong> files with `*.module.ts`
+- <strong>Providers:</strong> files with `*.service.ts`
+- <strong>Controllers:</strong> files with `*.controller.ts`
+- <strong>Model:</strong> files with `*.model.ts`
 
-`yarn run test`
+And following DDD we include this concepts:
+- <strong>Domain:</strong> files with `*.domain.ts`
+- <strong>Repository:</strong> files with `*.repository.ts`
 
-# End to End Testing
+So, example of module "File" `src/modules/file`:
```

Comment:
"As an example, a 'file' module would be:
- controller: file.controler.ts endpoints for exposing the business logic of the file module. 
- services: file.service.ts contains all the use-cases that are related to the file domain: moving a file, removing a file, etc."